pkg/asset/installconfig/gcp/validation.go

```diff
@@ -45,24 +45,31 @@ func Validate(client API, ic *types.InstallConfig) error {
 	allErrs = append(allErrs, validateNetworkProject(client, ic, field.NewPath("platform").Child("gcp"))...)
 	allErrs = append(allErrs, validateRegion(client, ic, field.NewPath("platform").Child("gcp"))...)
 	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("gcp"))...)
+	allErrs = append(allErrs, validateZones(client, ic)...)
 	allErrs = append(allErrs, validateInstanceTypes(client, ic)...)
 	allErrs = append(allErrs, validateCredentialMode(client, ic)...)
 
 	return allErrs.ToAggregate()
 }
 
 // ValidateInstanceType ensures the instance type has sufficient Vcpu and Memory.
-func ValidateInstanceType(client API, fieldPath *field.Path, project, zone, instanceType string, req resourceRequirements) field.ErrorList {
+func ValidateInstanceType(client API, fieldPath *field.Path, project string, zones sets.Set[string], instanceType string, req resourceRequirements) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	typeMeta, err := client.GetMachineType(context.TODO(), project, zone, instanceType)
-	if err != nil {
-		if _, ok := err.(*googleapi.Error); ok {
-			return append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
+	var typeMeta *compute.MachineType
+	for zone := range zones {
+		var err error
+		typeMeta, err = client.GetMachineType(context.TODO(), project, zone, instanceType)
+		if err != nil {
+			var gErr *googleapi.Error
```

Comment:
shouldn't this be the other way round? i.e, if there is no match for any instance, then return error, but if there is a match, exit out of the loop and continue?

Reply:
The Installer spreads the nodes through the availability zones: https://github.com/openshift/installer/blob/master/pkg/asset/machines/gcp/machines.go#L44-L46 so I find it safer to make sure that the instance is available in all zones. It was happening to me that the Installer was assigning a node to `us-central1-c` which is exactly the one that doesn't contain a `t2a-standard-*` type, even though `-a`, and `-b` do. With your suggestion I would've exited the loop on the match of zone `us-central1-a` but still would've failed during install.

Another possibility would be to replicate the way the machines are assigned zones by sorting the zones and then just checking for `len(nodes)` zones.

Reply:
so with the current implementation ,it would mean if you use `t2a-standard`, you'll have to specify the zone in the install-config, else it will fail the validation if it uses the default set of zones?

Reply:
Yes. But we might be able to change that with https://github.com/openshift/installer/pull/7096#issuecomment-1511004714